apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -1081,6 +1081,37 @@ function handleInlineHelpButton( calypsoPort ) {
 	);
 }
 
+/**
+ * If WelcomeTour is set to show, check if the App Banner is visible.
+ * If App Banner is visible, we set the Welcome Tour to not show.
+ * When the App Banner gets dismissed, we set the Welcome Tour to show.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function getIsAppBannerVisible( calypsoPort ) {
+	const isWelcomeGuideShown = select( 'automattic/wpcom-welcome-guide' ).isWelcomeGuideShown();
+	if ( ! isWelcomeGuideShown ) {
+		return;
+	}
+
+	const { port1, port2 } = new MessageChannel();
+	calypsoPort.postMessage(
+		{
+			action: 'getIsAppBannerVisible',
+			payload: {},
+		},
+		[ port2 ]
+	);
+	port1.onmessage = ( { data } ) => {
+		const { isAppBannerVisible, hasAppBannerBeenDismissed } = data;
+		if ( hasAppBannerBeenDismissed ) {
+			dispatch( 'automattic/wpcom-welcome-guide' ).setShowWelcomeGuide( true );
+		} else if ( isAppBannerVisible ) {
+			dispatch( 'automattic/wpcom-welcome-guide' ).setShowWelcomeGuide( false );
+		}
```

Comment:
Just noting that the action `setShowWelcomeGuide` does an API request to save the status of the tour (so it persists across sessions). I think this is fine for every case, but maybe confirm anyways that this is what we want 😃

Reply:
The only problem that I see is:
- app banner is visible
- show is set to false, and status is saved
- user closes the page, without dismissing the app banner
- user reopens -> show is set to false
- app banner is dismissed -> no welcome tour showed

It could be worth having another action that doesn't do an API request!

Reply:
This and the others should be solved in 4396643